app/src/lib/git/log.ts

```diff
@@ -155,9 +154,21 @@ export async function getCommits(
   const parsed = parse(result.stdout)
 
   return parsed.map(commit => {
-    const tags = getCaptures(commit.refs, /tag: ([^\s,]+)/g)
-      .filter(i => i[0] !== undefined)
-      .map(i => i[0])
+    // Ref is of the format: (HEAD -> master, tag: some-tag-name, tag: some-other-tag,with-a-comma, origin/master, origin/HEAD)
+    // Refs are comma separated, but some like tags can also contain commas in the name, so we split on the pattern ", " and then
+    // check each ref for the tag prefix. We used to use the regex /tag: ([^\s,]+)/g)`, but will clip a tag with a comma short.
+    const refs = commit.refs.split(', ')
+    const tags = []
+    for (let i = 0; i < refs.length; i++) {
+      const ref = refs[i]
```

Comment:
Nit:
```suggestion
    for (const ref of refs) {
```

The whole `for` could also be replaced with a `refs.flatMap(…)` returning `[]` for refs that don't start with `tag: `, but maybe it's not as straightforward as this? 🤔

Reply:
:D I do like the straight forward reading of the `for` loop, but I do really like the cleanliness of the one-liner. :P I think with the comment explaining what a ref can look like the `flatMap` is discernible enough. Thank you for the suggestion.